ffi/capi/src/neo_datetime.rs

```diff
@@ -540,10 +799,67 @@ pub mod ffi {
         }
     }
 
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::datetime::DateTimeFormatter, Struct)]
+    #[diplomat::rust_link(
+        icu::datetime::fieldsets::enums::DateAndTimeFieldSet::zone,
+        FnInStruct,
+        hidden
+    )]
+    #[diplomat::rust_link(icu::datetime::fieldsets::DT::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::MDT::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::YMDT::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::DET::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::MDET::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::YMDET::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::ET::zone, FnInStruct, hidden)]
+    #[diplomat::attr(demo_gen, disable)] // constructors are on a different type :(
```

Comment:
can we have a constructor for this at all?

please add docs to this type on how to construct it. Linking to rust docs is not sufficient here

Reply:
ctors could instead take DateTimeFormatter by reference. Not opposed, naming becomes `from` instead of `with`

Might feel less cluttered, instead of five hundred APIs on one type we split them across two.

Reply:
Currently we have:

`DateTimeFormatter::ymdt(...).withZoneGenericLong(...)`

I'm okay changing this to:

`ZonedDateTimeFormatter::fromDateTimeFormatterWithZoneGenericLong(..., DateTimeFormatter::ymdt(...))`

Is that what you had in mind?

Reply:
Would you like me to switch this over to constructors? @Manishearth

Reply:
Very slight preference for it, but I don't mind the current model too

Reply:
Very slight preference for it, but I don't mind the current model too

Reply:
https://github.com/unicode-org/icu4x/issues/5940#issuecomment-2722698945